lib/src/track/local/local.dart

```diff
@@ -154,6 +154,13 @@ abstract class LocalTrack extends Track {
         if (options.selfBrowserSurface != null) {
           constraints['selfBrowserSurface'] = options.selfBrowserSurface!;
         }
+
+        // Remove resolution settings to fix low-resolution screen share on Safari 17.
+        // related bug: https://bugs.webkit.org/show_bug.cgi?id=263015
+        if (lkBrowser() == BrowserType.safari &&
+            lkBrowserVersion().major == 17) {
```

Comment:
good thinking to specifically fix it for this Safari version, too. 
I think we should do something similar on the JS side. 
The bugreport in webkit only talks about `width: { max: ` constraints. maybe `width: { ideal: ` would be possible as a workaround for Safari 17. 
I remember it not being a valid workaround in general because `ideal` was causing issues in other Safari versions :/